### PR TITLE
[SDK] Implement missing C++ features for BtrFS

### PR DESCRIPTION
**Pull request by HeisSpiter**

### CODE REVIEW

**Comment by BenNottelling** [comment]
If you're interested at all for an STLport alternative this might be worth considering
https://github.com/msharov/ustl
It's looks like it was maintained for a while and it is much more updated than STLport which last got updated in 2008
Or it might be a worthhwhile source. Just putting that out there :)

File: sdk/include/c++/stlport/stl/_vector.h [hunk 1]
```diff
@@ -210,6 +210,8 @@ class vector : protected _STLP_PRIV _Vector_base<_Tp, _Alloc>
   reference at(size_type __n) { _M_range_check(__n); return (*this)[__n]; }
   const_reference at(size_type __n) const { _M_range_check(__n); return (*this)[__n]; }
 
+  _Tp* data()                   { return this->_M_start; }
```
Comment (learn-more) [inline sdk/include/c++/stlport/stl/_vector.h]:
```suggestion
  _Tp* data()                   { return this->_M_start; }
  const _Tp* data() const       { return this->_M_start; }
```

**Comment by HeisSpiter** [comment]
@SergeGautherie can you test if the produced shellbtrfs.dll produced with MSVC and this PR is working fine?

**Comment by zefklop** [comment]
looks fine to me

**Comment by SergeGautherie** [comment]
> Ideally, on Windows 2K3 or newer

I'll try, but be aware that so far my results with upstream (MSVC) binaries are:
WXP: seems to work (somewhat), WS03: no usable volume, Vista+: I don't have.

**Comment by HeisSpiter** [comment]
Then, go for a test on WXP @SergeGautherie 
Thanks for your review @zefklop :-)

**Comment by Neustradamus** [comment]
Dear @reactos team, @BelleNottelling, @zefklop, @SergeGautherie, @learn-more,

Since several years, I search to talk with @HeisSpiter, have you some news from him?

**Comment by binarymaster** [comment]
@Neustradamus this is not the right place to ask, use the chat for that: https://chat.reactos.org/

**Pull request merged**